src/server/rest/v1/service/RegistrationTokenService.ts

```diff
@@ -58,6 +58,16 @@ export default class RegistrationTokenService {
         user: req.user
       });
     }
+    // Check site is provided for site admins
+    if (Authorizations.isSiteAdmin(req.user) && !filteredRequest.siteAreaID) {
+      throw new AppError({
+        source: Constants.CENTRAL_SERVER,
+        errorCode: HTTPError.BAD_REQUEST,
```

Comment:
Use FORBIDDEN

Reply:
I will replace with this :

`if (Authorizations.isSiteAdmin(req.user) && !filteredRequest.siteAreaID) {`
`      throw new AppAuthError({`
`        errorCode: HTTPAuthError.FORBIDDEN,`
`        user: req.user,`
`        action: Action.CREATE, entity: Entity.TOKEN,`
`        module: MODULE_NAME, method: 'handleCreateRegistrationToken',`
`      });`
`    }`

But i thought it was more of a bad request as it misses the siteID for this request with the siteID user, it's not handled like user hasn't the rights as a "!Authorizations.canCreateRegistrationToken()"

Reply:
In that case put 2 'if' statements, one for the Auth and one for the missing param.
For the missing param, today we return a general exception (500), eg: assertIdIsProvided()...
This has to be changed everywhere if we go for that error code (BAD_REQUEST) for missing params :-)

Reply:
Ok i understood !! I made the changes for Forbidden hope it's ok now